nav2_collision_monitor/src/source.cpp

```diff
@@ -69,6 +69,12 @@ void Source::getCommonParameters(std::string & source_topic)
   nav2_util::declare_parameter_if_not_declared(
     node, source_name_ + ".enabled", rclcpp::ParameterValue(true));
   enabled_ = node->get_parameter(source_name_ + ".enabled").as_bool();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, source_name_ + ".source_timeout",
+    rclcpp::ParameterValue(source_timeout_.seconds()));      // node source_timeout by default
```

Comment:
which calls into question this -- can you test if this actually is not just full whole seconds?

Reply:
I swear it is not. This can be confusing as message header `stamp` is split between full seconds and full nanoseconds int: https://docs.ros2.org/latest/api/builtin_interfaces/msg/Time.html
But that is not the case for `rclcpp::Duration` and `rclcpp::Time` `seconds()` methods